Implement voting for session proposals

VelizarHristov opened the pull request:
Closes https://github.com/LWCW-Europe/scheduling-app/issues/29

VelizarHristov commented:
The only problem I'm aware of is that switching your vote to a different one might cause weird UI behaviour, especially if you do it multiple times quickly. 

The current merge conflicts are pointless, I'll resolve them after the code review or after more meaningful merge conflicts arise.

omarkohl commented on README.md:
```diff
@@ -119,14 +119,28 @@ You also need to add a SessionProposals table to your Airtable base:
     "Allow linking to multiple records" **checked**
   - `title`: Single line text (Primary field)
+  - `votes`: Link to another record (Votes)
+  - `votesCount`: Count (votes)
```
Why track deduplicate the votesCount? It can be calculated by counting the number of entries in the Votes table.

VelizarHristov replied:
It's a derived field, so it's used to simplify querying AirTable. It should be impossible to manually set it. Also I think that the `votes` field is required for this to work: note that I am not sending the votes to the client-side (which could potentially be a lot of data, for all proposals).

omarkohl commented:
@VelizarHristov why are votes linked to the event? Votes are linked to proposals and the proposals are linked to events so it seems redundant.

omarkohl commented:
I rebased this on top of my open PRs to be able to take advantage of the new DB migrations. I also added such a migration. I'll have a closer look once those are merged.

An initial test was successful, cool stuff!

Note there is one open TODO mentioned in the Quick Voting screen (link to overview).

VelizarHristov commented:
> I rebased this on top of my open PRs to be able to take advantage of the new DB migrations. I also added such a migration. I'll have a closer look once those are merged.
> 
> An initial manual test was successful, cool stuff!
> 
> Note there is one open TODO mentioned in the Quick Voting screen (link to overview).

Oops... there is not one but four TODOs, and I don't think I meant to leave any.

VelizarHristov commented:
> > I rebased this on top of my open PRs to be able to take advantage of the new DB migrations. I also added such a migration. I'll have a closer look once those are merged.
> > An initial manual test was successful, cool stuff!
> > Note there is one open TODO mentioned in the Quick Voting screen (link to overview).
> 
> Oops... there is not one but four TODOs, and I don't think I meant to leave any.

Now resolved. Two of them were already solved, I had just forgotten to remove the TODO comments; the link which shows up when the user has already voted on everything is not pretty, but it is functional, and I'd appreciate it if someone makes it prettier, or alternatively we could get rid of it since the back link is already on the top of the page and should be already familiar to the users.

omarkohl commented:
Two issues I found:

The link from quick voting does not look good on hover.

<img width="307" height="194" alt="image" src="https://github.com/user-attachments/assets/5729160b-c5d9-4d88-8d3a-9d9f25055fbc" />

On the proposal detail page it's not possible to view my current vote or to vote. All voting buttons are greyed out.

VelizarHristov commented:
> On the proposal detail page it's not possible to view my current vote or to vote. All voting buttons are greyed out.

Oops, I didn't even implement voting on the detail page - my bad. 

But I will have a LOT of merge conflicts to resolve when the redesign PR gets merged, so I will wait for that before I fix anything on this branch.

VelizarHristov commented:
> Two issues I found:
> 
> The link from quick voting does not look good on hover.
> 
> Start screenshot.
> <img alt="image" width="307" height="194" src="https://private-user-images.githubusercontent.com/1420656/476200679-5729160b-c5d9-4d88-8d3a-9d9f25055fbc.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ3NDA4ODksIm5iZiI6MTc1NDc0MDU4OSwicGF0aCI6Ii8xNDIwNjU2LzQ3NjIwMDY3OS01NzI5MTYwYi1jNWQ5LTRkODgtOGQzYS05ZDlmMjUwNTVmYmMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MDlUMTE1NjI5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmQyY2I0ZGMwOTQ2ODg3MTljNmM1ZjE0OTAxOGY4MTc4ZDEwNjBlMGRjODE5Nzk3N2JhN2M1MDM0MGM0YjBmMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.BXmIWgW54CdokWJMFIWZSgl1i8QkfZhkMoSN2y9tLfw">
> 
> End screenshot.
> 
> On the proposal detail page it's not possible to view my current vote or to vote. All voting buttons are greyed out.

I ended up just removing that link - the page's back buttons are right above anyway. I originally made it like that on purpose because I didn't know how to make it look nice.

omarkohl commented:
When changing the vote on the overview page the background for the old votes does not change. When reloading the page only the correct (last) vote remains, as expected.

<img width="265" height="251" alt="image" src="https://github.com/user-attachments/assets/7c9d91b6-0817-4075-b1d5-ff9e190565a1" />
